crates/uv-types/src/build_options.rs

```diff
@@ -211,6 +211,29 @@ impl NoBuild {
     }
 }
 
+#[derive(Debug, Default, Clone, Hash, Eq, PartialEq)]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+pub enum IndexStrategy {
+    /// Only use results from the first index that returns a match for a given package name.
+    ///
+    /// While this differs from pip's behavior, it's the default index strategy as it's the most
+    /// secure.
+    #[default]
+    FirstMatch,
+    /// Search for every package name across all indexes, exhausting the versions from the first
+    /// index before moving on to the next.
+    ///
+    /// In this strategy, we look for every package across all indexes. When resolving, we attempt
+    /// to use versions from the indexes in order, such that we exhaust all available versions from
+    /// the first index before moving on to the next. Further, if a version is found to be
+    /// incompatible in the first index, we do not reconsider that version in subsequent indexes,
+    /// even if the secondary index might contain compatible versions (e.g., variants of the same
+    /// versions with different ABI tags or Python version constraints).
```

Comment:
It might be worth ~one sentence mentioning that this risks dependency confusion attacks so that the "unsafe" in the name is justified here. (I know the PEP link gives the full context, but I think a one sentence call out here is important.)

(I'm assuming this is user facing docs. If not, the PEP link is enough IMO.)